### DISCUSSION TITLE
How to get the value of textarea/stage element via jquery correctly?

### DISCUSSION THREAD

**Discussion by vnali** [craftcms/cms#13310] [Q&A] [answered]
I'm trying to understand how setting/getting the value of a textarea works.

1- there is a textarea inside an element. i can see a stage element for this textarea in firefox inspector.
2- as long as I update text inside textarea via javascript and something like `$('#textareaField').text(val)`, i can see the latest changes inside textarea input, textarea element and stage element and I can get the latest updates via `$('#textareaField').text(val)` correctly.
3- but if i edit teaxtarea input manually by typing inside it, i can see the stage element is updated to the latest edit, but the textarea element is not updated.
4- after step 3, textarea element is not synced with the textarea input and stage element anymore.
for example if i update textarea via `$('#textareaField').text(val)`, this update only apply to textarea element, not textarea input or stage element

why textarea input, textarea element and stage element are not synced always?
what is the correct way to always set and get the latest update of a textarea via jquery?

**Comment by brandonkelly** [comment, marked as answer]
This is more of a general JavaScript/HTML question. Craft doesn’t do anything special here. The `<textarea>` is just a `<textarea>`.

`.text()` is for getting the HTML-encoded textual value of an element, or setting a value that should become HTML-encoded.

To get the value of a textarea, you should use `.val()` instead (`('#textareaField').val()`).
